pkg/common/cns-lib/volume/manager.go

```diff
@@ -1530,3 +1534,134 @@ func (m *defaultManager) QuerySnapshots(ctx context.Context, snapshotQueryFilter
 	}
 	return resp, err
 }
+
+func (m *defaultManager) CreateSnapshot(ctx context.Context, volumeID string, desc string) (string, string, string, *time.Time, error) {
+	log := logger.GetLogger(ctx)
+	err := validateManager(ctx, m)
+	if err != nil {
+		return "", "", "", nil, err
+	}
+	// Set up the VC connection
+	err = m.virtualCenter.ConnectCns(ctx)
+	if err != nil {
+		log.Errorf("ConnectCns failed with err: %+v", err)
+		return "", "", "", nil, err
+	}
+
+	// Construct the CNS SnapshotCreateSpec list
+	var cnsSnapshotCreateSpecList []cnstypes.CnsSnapshotCreateSpec
+	cnsSnapshotCreateSpec := cnstypes.CnsSnapshotCreateSpec{
+		VolumeId: cnstypes.CnsVolumeId{
+			Id: volumeID,
+		},
+		Description: desc,
```

Comment:
Do you think we should sanitize desc?, are there limits on the length of description?

Reply:
I don't think so, snapshot description, desc, used in vSphere CSI comes from external snapshotter. It comes with the format, "snapshot-<UUID>". So the length of the description can not be too long.